keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModelBuilder.java

```diff
@@ -118,8 +136,12 @@ public RegressionModelBuilder withSampling(SamplingModelFitting sampling) {
             getWeightsVertex()
         );
 
-        performDataFitting(regressionGraph, outputTrainingData);
-        return new RegressionModel<>(regressionGraph);
+        ModelFitter<DoubleTensor, OUTPUT> fitter = samplingAlgorithm == null ?
+            this.regularization.createFitterForGraph(regressionGraph) :
+            samplingAlgorithm.createFitterForGraph(regressionGraph);
+
+        regressionGraph.observeValues(inputTrainingData, outputTrainingData);
+        return new RegressionModel<>(regressionGraph, fitter);
```

Comment:
The `fitter` contains the `regressionGraph`, so it doesn't feel quite right that you have to pass in both the `regressionGraph` and the `fitter` here.

I wonder if the `ModelFitter` API should actually be `fit(ModelGraph)` instead of `fit()`. That would make this cleaner and you wouldn't be storing the `regressionGraph` in 2 different places.